### PR TITLE
[TESTKIT] Reintroduce old code and mark them obsolete

### PR DESCRIPTION
**Pull request by Arkatufus**
Reintroduce `Dispose` and `AfterAll` method to `TestKit`

### CODE REVIEW

File: src/contrib/testkits/Akka.TestKit.Xunit2/TestKit.cs [hunk 1]
```diff
@@ -168,7 +179,24 @@ public virtual Task InitializeAsync()
...
+        /// Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources.
+        /// </summary>
+        /// <param name="disposing">
+        /// if set to <c>true</c> the method has been called directly or indirectly by a  user's code.
+        /// Managed and unmanaged resources will be disposed.<br /> if set to <c>false</c> the method
+        /// has been called by the runtime from inside the finalizer and only unmanaged resources can
+        ///  be disposed.
+        /// </param>
+        [Obsolete("Dispose(bool) is deprecated, please use DisposeAsync() instead")]
+        protected virtual void Dispose(bool disposing)
```
Comment (Aaronontheweb) [inline src/contrib/testkits/Akka.TestKit.Xunit2/TestKit.cs:198]: lgtm

Comment (Aaronontheweb) [reply src/contrib/testkits/Akka.TestKit.Xunit2/TestKit.cs:198]: Don't we still need to implement `IDisposable` on this class?

Comment (Aaronontheweb) [reply src/contrib/testkits/Akka.TestKit.Xunit2/TestKit.cs:198]: Yes, we do https://github.com/akkadotnet/akka.net/blob/a45501991bd366a4aa7db60347760332b6cd30c4/src/contrib/testkits/Akka.TestKit.Xunit2/TestKit.cs#L22

**Review by Aaronontheweb — CHANGES_REQUESTED** [review]
Still need to add `IDisposable` back and do API approvals

File: src/contrib/testkits/Akka.TestKit.Xunit2/TestKit.cs (hunk 1 above)
Comment (Arkatufus) [reply src/contrib/testkits/Akka.TestKit.Xunit2/TestKit.cs:198]: Done

**Review by Aaronontheweb — APPROVED** [review]

**Review by Aaronontheweb — APPROVED** [review]
LGTM

**Pull request merged**